sklearnex/spmd/basic_statistics/tests/test_incremental_basic_statistics_spmd.py

```diff
@@ -0,0 +1,303 @@
+            weights, sycl_queue=queue, target_df=dataframe
+        )
+        local_weights = _get_local_tensor(weights)
+        split_local_weights = np.array_split(local_weights, num_blocks)
+
+    option, _, _ = result_option
+    incbs_spmd = IncrementalBasicStatistics_SPMD(result_options=option)
+    incbs = IncrementalBasicStatistics(result_options=option)
+
+    for i in range(num_blocks):
+        local_dpt_data = _convert_to_dataframe(
+            split_local_data[i], sycl_queue=queue, target_df=dataframe
+        )
+        if weighted:
+            local_dpt_weights = _convert_to_dataframe(
+                split_local_weights[i], sycl_queue=queue, target_df=dataframe
+            )
+            incbs_spmd.partial_fit(local_dpt_data, sample_weight=local_dpt_weights)
+        else:
+            incbs_spmd.partial_fit(local_dpt_data)
+
+    if weighted:
+        incbs.fit(dpt_data, sample_weight=dpt_weights)
+    else:
+        incbs.fit(dpt_data)
+
+    assert_allclose(getattr(incbs_spmd, option), getattr(incbs, option))
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("num_blocks", [1, 2])
+@pytest.mark.parametrize("weighted", [True, False])
+@pytest.mark.parametrize("n_samples", [100, 10000])
+@pytest.mark.parametrize("n_features", [10, 100])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.mpi
+def test_incremental_basic_statistics_partial_fit_spmd_synthetic(
+    dataframe, queue, num_blocks, weighted, n_samples, n_features, dtype
+):
+    # Import spmd and batch algo
+    from sklearnex.basic_statistics import IncrementalBasicStatistics
+    from sklearnex.spmd.basic_statistics import (
+        IncrementalBasicStatistics as IncrementalBasicStatistics_SPMD,
+    )
+
+    tol = 2e-3 if dtype == np.float32 else 1e-7
+
+    # Create gold data and process into dpt
+    data = _generate_statistic_data(n_samples, n_features, dtype=dtype)
+    local_data = _get_local_tensor(data)
+    split_local_data = np.array_split(local_data, num_blocks)
+    split_data = np.array_split(data, num_blocks)
+
+    if weighted:
+        weights = _generate_weights(n_samples, dtype=dtype)
+        local_weights = _get_local_tensor(weights)
+        split_local_weights = np.array_split(local_weights, num_blocks)
+        split_weights = np.array_split(weights, num_blocks)
+
+    incbs_spmd = IncrementalBasicStatistics_SPMD()
+    incbs = IncrementalBasicStatistics()
+
+    for i in range(num_blocks):
+        local_dpt_data = _convert_to_dataframe(
+            split_local_data[i], sycl_queue=queue, target_df=dataframe
+        )
+        dpt_data = _convert_to_dataframe(
+            split_data[i], sycl_queue=queue, target_df=dataframe
+        )
```

Comment:
could you add ``` assert(dpt_data.dtype == dtype) ``` somewhere later in your tests, since dtype isn't being passed to _convert_to_dataframe, I would like to verify that the conversion hasn't corrupted the dtype. Either that or explicitly pass the dtype as a kwarg.

Reply:
tbh i think it should be verified in separate _convert_to_dataframe test, not here. and this construction used a lot of times in our tests and i've never seen dtype is provided

Reply:
https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/tests/_utils.py#L307

This is a pattern used extensively in test_patching and test_run_to_run_stability.  This will pass the dtype setting to the dataframe constructor to insure proper dtyping.  I would like your conversions to be explicit in the dtype it requests, it isn't clear to me if it is guaranteed to convert to the same datatype.

Reply:
but there are a lot of other places in the repo where it is not specified and it works well. I still think that verifying of _convert_to_dataframe work is not the goal of this pr and should be resolved separately

Reply:
There is already a circumstance in this PR where variation in the datatype has wide impact on the accuracy of the resultant data.  That could be due to an issue in oneDAL, sklearnex interface, and/or within the tests themselves.  By explicitly typing your conversions, you are being clear to the user, developer, and reviewer the chain of a vital parameter in testing and saving effort for the future ticket.  There is good logical reasoning as to why I am asking for this, precedence is an insufficient argument in this case.

Reply:
> could you add `assert(dpt_data.dtype == dtype)` somewhere later in your tests, since dtype isn't being passed to _convert_to_dataframe, I would like to verify that the conversion hasn't corrupted the dtype. Either that or explicitly pass the dtype as a kwarg.

It is not recommended to explicitly add dtype, shape or other kwargs for the _convert_to_dataframe. It is expected that it will work as is, since all primitives inside uses input object metadata and attributes. Actually passing numpy dtype as a dtype param can break here for non numpy dataframes.
I think, this is not a scope for the current PR. We can separately cover with tests `_convert_to_dataframe` for this kind of checks (just needed a ticket now). @icfaust if there some bugs that you investigated related this please also report it as a bug for future investigation and work.

Reply:
> > could you add `assert(dpt_data.dtype == dtype)` somewhere later in your tests, since dtype isn't being passed to _convert_to_dataframe, I would like to verify that the conversion hasn't corrupted the dtype. Either that or explicitly pass the dtype as a kwarg.
> 
> It is not recommended to explicitly add dtype, shape or other kwargs for the _convert_to_dataframe. It is expected that it will work as is, since all primitives inside uses input object metadata and attributes. Actually passing numpy dtype as a dtype param can break here for non numpy dataframes. I think, this is not a scope for the current PR. We can separately cover with tests `_convert_to_dataframe` for this kind of checks (just needed a ticket now). @icfaust if there some bugs that you investigated related this please also report it as a bug for future investigation and work.

Please see the codebase: https://github.com/intel/scikit-learn-intelex/blob/main/onedal/tests/utils/_dataframes_support.py#L139 Issues related to dtype changing  have already been addressed, and are being used in testing all estimators and methods as per test_patching.py . There is a known issue that I have had @olegkkruglov make a ticket for here: https://github.com/intel/scikit-learn-intelex/pull/1961#discussion_r1734356127 which is correlated to the dtype of the input.  Since I am outvoted, you can remove my changes requested, but I only ask that someone else beyond @bdmoore1 provides an approval before merging.

Reply:
I vote also for checking this.
I suggest:
1) I think it would be better just to add some asserts after `_convert_to_dataframe` call to check that returned object has the same dtype. But it also somewhere should be refactored and added for other algos.
2) Also good to have a ticket for covering the `_convert_to_dataframe` with unit tests.


Generally, I think we should avoid explicit shape, dtype, or other attr providing. I expect that all dataframes construct methods called for the input object handling all attributes and metadate, but make sense to check as proposed on 2).